apis/v1alpha2/validation/validation.go

```diff
@@ -34,6 +36,23 @@ var (
 	}
 )
 
+// ValidateGatewayClassName validates that the given name can be used as
+// a GatewayClass name.
+var ValidateGatewayClassName = apivalidation.NameIsDNSSubdomain
+
+// ValidateGatewayClass validates gc according to the Gateway API specification.
+// For additional details of the GatewayClass spec, refer to:
+// https://gateway-api.sigs.k8s.io/spec/#networking.x-k8s.io/v1alpha2.GatewayClass
+func ValidateGatewayClass(gc *gatewayv1a2.GatewayClass) field.ErrorList {
+	return validateGatewayClassObjMeta(&gc.ObjectMeta, field.NewPath("metadata"))
+}
+
+// validateGatewayClassObjMeta validates whether required fields of metadata are set according
+// to the Gateway API specification.
+func validateGatewayClassObjMeta(meta *metav1.ObjectMeta, path *field.Path) field.ErrorList {
+	return apivalidation.ValidateObjectMeta(meta, false, ValidateGatewayClassName, path)
+}
```

Comment:
Please also add this validation to `ValidateGatewayClassUpdate()` code path.

Reply:
Is a name change even possible?

Reply:
Probably not. The main motivation for this was to have most validation logic between create and update as common and only some corner cases are handled specially.